src/modules/roc_packet/target_stdio/roc_packet/ip_to_str.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_packet/target_stdio/roc_packet/ip_to_str.h
+//! @brief Convert ip address to string.
+
+#ifndef ROC_PACKET_IP_TO_STR_H_
+#define ROC_PACKET_IP_TO_STR_H_
+
+#include "roc_core/noncopyable.h"
+#include "roc_packet/address.h"
+
+namespace roc {
+namespace packet {
+
+//! Format ip string of network address.
+class ip_to_str : public core::NonCopyable<> {
```

Comment:
I think it's an overkill to create a class for every Address field (ip, miface), especially given they're used only in one place. I suggest to remove ip_to_str and multicast_iface_to_str at all and just use get_ip() and get_multicast_iface() directly in UDPReceiver.

Reply:
Well, they are used at least in 2 places:

* udp_receiver_port - for logging
* udp_receiver_port - join/leave multicast group
* address_to_str - again for logging

As address_to_str may be used in many place I think it is worth to keep these helpers, probably we can use templates here to avoid adding helper for each `get*`.

Reply:
I don't suggest to remove address_to_str. I suggest to remove ip_to_str and multicast_iface_to_str.

They are used only UDPReceiverPort. And we can easily avoid them by using get_ip() directly:

```
char ip[64];
address_.get_ip(ip, sizeof(ip));

char miface[64];
address_.get_miface(miface, sizeof(miface));

if (int err = uv_udp_set_membership(&handle_, ip,
                                    miface,
                                    UV_JOIN_GROUP)) {
    roc_log(LogError, "udp receiver: uv_udp_set_membership(): [%s] %s",
            uv_err_name(err), uv_strerror(err));
    return false;
}

roc_log(LogDebug, "udp receiver: joined multicast group %s at interface %s",
        ip,
        miface);
```

ip_to_str and multicast_iface_to_str are just convinient wrappers primarily useful for logging. Such wrappers make sense when there are many usages. If there are very few usages, I'd prefer to avoid adding them.

Reply:
Done